src/pages/VKeywordCounter.vue

```diff
@@ -0,0 +1,274 @@
+<script setup lang="ts">
+import { ref, computed, onMounted, watch } from 'vue';
+import { MaCard, MaInput, MaButton } from '@mobileaction/action-kit';
+
+interface Props {
+  text?: string;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  text: ''
+});
+
+const inputText = ref('');
+const keywordStats = ref<{ keyword: string; count: number; percentage: number }[]>([]);
+const totalWords = ref(0);
+
+const processText = (text: string) => {
+  if (!text.trim()) return [];
+  
+  return text
+    .toLowerCase()
+    .replace(/[^\w\s]/g, ' ')
+    .replace(/\s+/g, ' ')
+    .trim()
+    .split(' ')
+    .filter(word => word.length > 0);
+};
+
+const updateWordCount = () => {
+  const words = processText(inputText.value);
+  totalWords.value = words.length;
+};
+
+onMounted(() => {
+  if (props.text) {
+    inputText.value = props.text;
+    updateWordCount();
+  }
+});
+
+watch(inputText, updateWordCount);
+
+const countWords = (words: string[]) => {
+  const wordCount: { [key: string]: number } = {};
+  words.forEach(word => {
+    wordCount[word] = (wordCount[word] || 0) + 1;
+  });
+  return wordCount;
+};
+
+const calculatePercentage = (count: number, total: number) => {
+  return Math.round((count / total) * 100 * 100) / 100;
+};
+
+const formatKeywordStats = (wordCount: { [key: string]: number }, totalWords: number) => {
+  return Object.entries(wordCount)
+    .map(([keyword, count]) => ({
+      keyword,
+      count,
+      percentage: calculatePercentage(count, totalWords)
+    }))
+    .sort((a, b) => b.count - a.count);
+};
+
+const analyzeKeywords = () => {
+  if (totalWords.value === 0) {
+    keywordStats.value = [];
+    return;
+  }
+
+  const words = processText(inputText.value);
+  const wordCount = countWords(words);
+  keywordStats.value = formatKeywordStats(wordCount, totalWords.value);
+};
+
+const hasResults = computed(() => keywordStats.value.length > 0);
+</script>
+
+<template>
+  <div class="keyword-counter-wrapper">
+    <div class="main-container">
+      <!-- Input Section -->
+      <div class="input-container">
+        <ma-card 
+          title="Keyword Counter"
+          description="Analyze keyword frequency and percentages in your text"
+          headerIcon="chart-bar"
+          class="h-full"
+        >
+          <div class="input-section">
+            <ma-input
+              v-model:value="inputText"
+              type="textarea"
+              title="Text Input"
+              placeholder="Enter your text here..."
+              hintText="Paste or type the text you want to analyze for keyword frequency"
+              class="mb-4"
+            />
+            
+            <ma-button 
+              type="primary" 
+              size="medium"
+              @click="analyzeKeywords"
+              class="w-full sm:w-auto"
+            >
```

Comment:
Events should be last
```suggestion
            <ma-button 
              type="primary" 
              size="medium"
              class="w-full sm:w-auto"
              @click="analyzeKeywords"
            >
```